packages/dd-trace/src/appsec/waf/waf_manager.js

```diff
@@ -51,19 +54,34 @@ class WAFManager {
     return wafContext
   }
 
-  update (newRules) {
-    try {
-      this.ddwaf.update(newRules)
+  update (product, rules, path) {
+    if (product === 'ASM_DD' && this.ddwaf.configPaths.includes(DEFAULT_WAF_CONFIG_PATH)) {
+      this.ddwaf.removeConfig(DEFAULT_WAF_CONFIG_PATH)
+    }
 
-      if (this.ddwaf.diagnostics.ruleset_version) {
-        this.rulesVersion = this.ddwaf.diagnostics.ruleset_version
-      }
+    const success = this.ddwaf.createOrUpdateConfig(rules, path)
+    const diagnostics = this.ddwaf.diagnostics
 
-      Reporter.reportWafUpdate(this.ddwafVersion, this.rulesVersion, true)
-    } catch (error) {
-      Reporter.reportWafUpdate(this.ddwafVersion, 'unknown', false)
+    if (diagnostics.ruleset_version) {
+      this.rulesVersion = diagnostics.ruleset_version
+    }
+
+    if (product === 'ASM_DD' && !success && !this.ddwaf.configPaths.some(cp => cp.includes('ASM_DD'))) {
```

Comment:
so currently you're doing "remove default rule, update, and if fail add back default rule" right ?
why not just do "add new rules, if success remove old ones" ?

Reply:
It is not possible with the current WAF.